added the drone security cheatsheet (#1612)
OWASP/CheatSheetSeries: cheatsheets/Drone_security_sheet.md

# Drone Security Cheat Sheet

## Introduction

Drone security is crucial due to their widespread adoption in industries such as military, construction, and community services. With the increasing use of drone swarms, even minor security lapses can lead to significant risks.

This cheat sheet provides an overview of vulnerable endpoints in drone systems and strategies to mitigate security threats.

---

## Drone System Components

A typical drone architecture consists of three main components:

1. **Unmanned Aircraft (UmA)** – The physical drone itself, including its sensors and onboard systems.
2. **Ground Control Station (GCS)** – The interface used to control and monitor drone operations.

> 3. **Communication Data-Link (CDL)** – The network connection between the drone and the GCS

Reviewer: ```suggestion
3. **Communication Data-Link (CDL)** – The network connection between the drone and the GCS.
```

- **Wi-Fi Weaknesses** – Weak authentication or unprotected channels can allow unauthorized access. This is even possible through simple [microcontrollers like ESP8266](https://github.com/SpacehuhnTech/esp8266_deauther)!

    - Use **802.11w MFP (Management Frame Protection)** to prevent Wi-Fi deauthentication attacks. Don't worry, if your Wi-Fi systems are up to date, then this is a default protocol now.

### 2. Authentication & Access Control

Most drone controllers use 2 sets of computers,

1. The main chip that performs the PID control and handles motors

2. An additional SoC (called the **companion computer**) to manage peripherals (like the cameras, LiDARs etc.) and send telemetry data.

Thus, it becomes very important to maintain their security as well. The possible risks in this case are:

- **Companion Computers** – Open ports (e.g., SSH, FTP) can be exploited if not securely configured.

- **User Error and Misconfiguration** – Misconfigured security settings can expose the drone to risks.

### 3. Physical Security

> If you're done is ever captured or lost, you should ensure that its not physically possible to steal data from it. This may happen under the following conditions:

Reviewer: ```suggestion
If your drone is ever captured or lost, you should ensure that its not physically possible to steal data from it. This may happen under the following conditions:
```

- **Insufficient Physical Security** – Unsecured USB ports or exposed hardware can lead to data theft or tampering.

- **Insecure Supply Chain** – Compromised components from suppliers can introduce hidden vulnerabilities.

- **End-of-Life Decommissioning Risks** – Improperly decommissioned drones may retain sensitive data or be repurposed maliciously.

### 4. Sensor Security

> With drones implementing control logic depending on how close they are to other drones or aerial vehicles, manipulating sensor data is extremely disastrous!

Reviewer: ```suggestion
With drones implementing control logic depending on how close they are to other drones or aerial vehicles, manipulating sensor data can be disastrous!
```

> Yes, attackers can manipulate drone sensors (GPS, cameras, altimeters) to feed incorrect data. Think of this more like how [struxnet](https://en.wikipedia.org/wiki/Stuxnet) changed the temperatures of the Uranium holdings in Iran.

Reviewer: ```suggestion
Attackers can manipulate drone sensors (GPS, cameras, altimeters) to feed incorrect data. Think of this like how [stuxnet](https://en.wikipedia.org/wiki/Stuxnet) changed the speed of the Uranium centrifuges in Iran while still reporting the speed as normal.
```

To prevent this, there is new research being developed involving **watermarked signals** whose **entropy** can be used to determine if the sensor values are correct of not. Read more about this method [here](https://ieeexplore.ieee.org/abstract/document/9994719).

### 5. Logging & Monitoring

- **Inadequate Logging and Monitoring** – Without sufficient monitoring, security breaches or operational anomalies may go undetected.

> - **Integration Issues** – The cameras require webserver configurations, and if poorly integrated, these web servers on cameras or telemetry systems may expose vulnerabilities that can be used to gather sensitive information.

Reviewer: ```suggestion
- **Integration Issues** – Some cameras require webserver configurations, and if poorly integrated, these web servers on cameras or telemetry systems may expose vulnerabilities that can be used to gather sensitive information.
```

> To prevent this, ensure that your credentials are strong! Additionally you may choose to implement SecOps softwares like [DefectDojo](https://github.com/DefectDojo/django-DefectDojo) for continous monitoring of your systems.

Reviewer: Isn't DefectDojo for vulnerability tracking? How does this help with monitoring? Does DefectDojo have monitoring capabilities as well?

Author: No, it doesn't. I got carried away, this was an oversight on my part.

---

## Secure Communication Protocols

Below are some protocols used by drone systems to communicate. This can be either between each other (if in a horde) or with the ground stations. We have mentioned what can go wrong with each protocol and also provided recommendations.

1. **MAVLink 2.0** – A widely used protocol for communication between drones and ground control stations (GCS).

   - Implement **message signing** to prevent spoofing and replay attacks.

   - You must secure **heartbeat messages** to avoid [command injection vulnerabilities](https://owasp.org/www-community/attacks/Command_Injection). A heartbeat message is usually a single byte that is sent at a certain frequency to all other nodes, informing of the device's existence. The frequency is important here!

   - Tools like **ArduPilot** and **PX4** support MAVLink 2.0 security enhancements. There are thoroughly tested softwares and hence recommended.

   - Utilize **end-to-end encryption**! Either through TLS or DTLS is fine and good.

2. **CAN (Controller Area Network) Bus** – A communication protocol used between internal drone system components (e.g., flight controllers, ESCs, GPS modules).

>    - Most attacks require **physical access** to exploit CAN. It works on differentially signals and hardware hacking may be possible by tapping into them.

Reviewer: What is meant by "differentially signals"?

By implementing these security measures, drone operators can significantly reduce the risks of cyberattacks and unauthorized access to UAV communication systems.

## Summary

The following table summaries the different attack vectors for a drone system.

| Attack |  | Targets | | | | | Security Measures | |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Type | Nature | Privacy| Data Confidentiality | Integrity | Accessibility | Authentication|Non-Cryptographic | Cryptographic |
| Malware | Infection | x | x |x |x |x | Control access, system integrity solutions and multi-factor authentication | Hybrid lightweight Intrusion Detection System |
| BackDoor Access | Infection |x|x|x| x|x | Multi-factor robust authentication scheme | Hybrid lightweight Intrusion Detection System, vulnerability assessment |
| Social Engineering | Exploitation | x|x |- |- |x | N/A | Raising awareness, training operators |
| Baiting | Exploitation |x| x| x|- |x | N/A | Raising awareness, training operators |
| Injection/Modification | Exploitation |x |- |x |- |- | Message authentication or digital signature | Machine-Learning hybrid Intrusion Detection System, timestamps |

> | Fabrication | Exploitation |x |- |x |- |x | Multi-factor authentication, message authentication or digital signature | , Assigning privilege |

Reviewer: ```suggestion
| Fabrication | Exploitation |x |- |x |- |x | Multi-factor authentication, message authentication or digital signature | Assigning privilege |
```
